modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/snapshot/GlobalSnapshotStateChannelEventsProcessor.scala

```diff
@@ -67,26 +66,64 @@ object GlobalSnapshotStateChannelEventsProcessor {
     new GlobalSnapshotStateChannelEventsProcessor[F] {
       private val logger = Slf4jLogger.getLoggerFromClass[F](GlobalSnapshotStateChannelEventsProcessor.getClass)
 
+      def deserialize[A: Decoder](binary: Signed[StateChannelSnapshotBinary]): F[Option[A]] =
+        jsonBrotliBinarySerializer.deserialize[A](binary.value.content).map(_.toOption)
+
+      def buildSnapshotFeesInfo(
+        lastGlobalSnapshotInfo: GlobalSnapshotInfo,
+        event: StateChannelOutput,
+        allFeesAddresses: Map[Address, Set[Address]]
+      ): F[SnapshotFeesInfo] =
+        event.snapshotBinary.value.lastSnapshotHash match {
+          case hash if hash == Hash.empty => SnapshotFeesInfo.empty.pure // genesis
+          case _ =>
+            deserialize[Signed[CurrencyIncrementalSnapshot]](event.snapshotBinary).flatMap {
+              case None => new Exception("Could not get snapshot after deserializing").raiseError[F, SnapshotFeesInfo]
+              case Some(snapshot) =>
+                Async[F].delay {
+                  val stakingBalance = fetchStakingBalance(event.address, lastGlobalSnapshotInfo)
+                  val sortedMessagesDesc = snapshot.value.messages.map(_.toList.sortBy(_.ordinal.value.value)(Ordering[Long].reverse))
```

Comment:
Alternatively you can just use `-` before the value:

```scala
val sortedMessagesDesc = snapshot.value.messages.map(_.toList.sortBy(-_.ordinal.value.value))
```

Reply:
👍🏼